compiler/rustc_target/src/abi/call/riscv.rs

```diff
@@ -87,7 +87,31 @@ where
             FieldsShape::Primitive => {
                 unreachable!("aggregates can't have `FieldsShape::Primitive`")
             }
-            FieldsShape::Union(_) => {
+            FieldsShape::Union(count) => {
+                if arg_layout.is_transparent() {
+                    let mut non_zst_elem = None;
+                    for i in 0..count.get() {
+                        let elem_layout = arg_layout.field(cx, i);
+                        if elem_layout.is_zst() {
+                            continue;
+                        }
+                        // There can only be one.
+                        if non_zst_elem.is_some() {
+                            return Err(CannotUseFpConv);
```

Comment:
```suggestion
                            bug!("two non-1-ZST fields in repr(transparent) union");
```

Reply:
`bug!` is not available in `rustc_target` (as it's defined in `rustc_middle` which is not depended on by `rustc_target`) so I used `unreachable!` instead.